app/modules/submissions/resources.py

```diff
@@ -161,22 +162,46 @@ def post(self, args):
         return submission
 
 
+@api.resolve_object_by_model(Submission, 'submission', return_not_found=True)
```

Comment:
Couldn't think of a better name so left as is, should this be first or last in the list of decorators?

Reply:
I can think of a gentle argument for either one? Probably lean towards last and let the login check happen first.

Reply:
tweaked as suggested, mostly, I left the responses at the end of the list as that seemed more intuitive